api/v1/atlasipaccesslist_types.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2025 MongoDB.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/api"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/api/v1/status"
+)
+
+func init() {
+	SchemeBuilder.Register(&AtlasIPAccessList{}, &AtlasIPAccessListList{})
+}
+
+// AtlasIPAccessListSpec defines the desired state of AtlasIPAccessList.
+type AtlasIPAccessListSpec struct {
+	// ProjectReference is the dual external or kubernetes reference with access credentials
+	ProjectDualReference `json:",inline"`
+	// Entry using an IP address in this access list entry.
+	// +optional
+	IPAddress string `json:"ipAddress,omitempty"`
+	// Range of IP addresses in CIDR notation in this access list entry.
+	// +optional
+	CIDRBlock string `json:"cidrBlock,omitempty"`
+	// Unique identifier of AWS security group in this access list entry.
+	// +optional
+	AwsSecurityGroup string `json:"awsSecurityGroup,omitempty"`
+	// Timestamp in ISO 8601 date and time format in UTC after which Atlas deletes the temporary access list entry.
+	// +optional
+	DeleteAfterDate *metav1.Time `json:"deleteAfterDate,omitempty"`
+	// Comment associated with this access list entry.
+	// +optional
+	Comment string `json:"comment,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +groupName:=atlas.mongodb.com
+// +kubebuilder:resource:categories=atlas,shortName=aial
+// +kubebuilder:printcolumn:name="Ready",type=string,JSONPath=`.status.conditions[?(@.type=="Ready")].status`
+
+// AtlasIPAccessList is the Schema for the atlasipaccesslists API.
+type AtlasIPAccessList struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   AtlasIPAccessListSpec          `json:"spec,omitempty"`
```

Comment:
The internal shape of this "List" strict is not a list, but a single entry. Is this pending to change?